features/support/step_errors.rb

```diff
@@ -0,0 +1,157 @@
+# Errors
+#  All errors raised by any of the steps defined in our code should be a
+#  subclass of SHFStepsError.  This allows us to discriminate and thus rescue
+#  error that arise from our code vs. all other errors.
+#
+#  Define any specific errors in this file.
+#
+#  See the RSpec in spec/features/support/cucumber_steps_errors_spec.rb for examples
+#
+# @ref NestedExceptions: http://wiki.c2.com/?NestedException
+
+
+# ---------------------------------------------
+# @module SHFMessageFormatter
+#
+# @description Helpers for formatting messages
+#
+module SHFMessageFormatter
+
+  # @return [String] - If str is empty, return an empty string, else return it with single quotes around it
+  def empty_or_quoted(str)
+    str.blank? ? '' : "'#{str}'"
+  end
+
+
+  # @return [String] -  If a string is not blank, put a space before it if space_before is true, else put the space after it
+  def space(str, space_before = true)
+    str.blank? ? '' : (space_before ? " #{str}" : "#{str} ")
+  end
+end
+
+
+
+# ---------------------------------------------
+#
+# @class SHFStepsError
+#
+# @description Parent class for all errors raised by any of our steps.
+#   This can track an 'original_error_message': an error that may have been raised
+#   this.  This can be used to help provide context when printing out the
+#   message for this error; it helps to keep errors from being 'swallowed'
+#   or disappeared during a chain of errors.
+#
+class SHFStepsError < StandardError
+  include SHFMessageFormatter
+
+  attr_reader :original_error_message
+
+  # If there is an error message in the global variable '$!' then this will
+  # use it. This enables the object to “magically” discover the originating
+  # exception message on initialization.
+  def initialize
+    super
+    @original_error_message = $! # This is the message of the last raised exception
+  end
```

Comment:
I suggest considering use of `Exception#cause` as a way to get information related to predecessor exceptions.  

This is cleaner (IMO) and leverages Ruby better.  It also allows for gathering information for any number of levels of predecessor exceptions.

I opened a PR - against your PR - where I made this change (and adjusted tests accordingly).

Your way here works fine, of course, but we are also establishing a pattern on how to do more sophisticated exception management, and as such this PR is important (I did happen to use `cause` for payments errors).

Here's a great (and short) write up on this: https://www.honeybadger.io/blog/nested-errors-in-ruby-with-exception-cause/

Reply:
Excellent suggestion.  I totally agree: we want this to work really well and `cause` is a better approach.  (Starr has posted lots of great info.  And they started Honeybadger in the neighborhood where I used to live. :-) )